presto-orc/src/main/java/com/facebook/presto/orc/DwrfStreamReorderingConfig.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.orc.proto.DwrfProto;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class DwrfStreamReorderingConfig
+{
+    // column is the logical columns index for a table with n columns
+    // This index starts from 0 to n - 1
+    Map<Integer, List<DwrfProto.KeyInfo>> columnIdToFlatMapKeyIds;
+    public DwrfStreamReorderingConfig(Map<Integer, List<DwrfProto.KeyInfo>> columnIdToFlatMapKeyIds) {
+        this.columnIdToFlatMapKeyIds = requireNonNull(columnIdToFlatMapKeyIds, "columnIdToFlatMapKeyIds cannot be null");
```

Comment:
Let's create an immutable copy of input.
Presto-orc usually uses `column` instead of `columnId`.

Reply:
Are you sure about creating a deep copy? It can be quite expensive from memory footprint point of view as the ordering config can have thousands of feature ID(s)?

Reply:
Makes sense.

Reply:
Yes, like that idea to change everything to immutable